CMakeLists.txt

```diff
@@ -92,12 +92,16 @@ set_property(CACHE USE_SMM PROPERTY STRINGS blas libxsmm)
 
 option(USE_CUDA "Build with CUDA support" OFF)
 option(USE_HIP "Build with HIP support" OFF)
-# USE_CUDA and USE_HIP are mutually exclusive options: we either compile with
-# nvcc OR with hipcc
-if (USE_CUDA AND USE_HIP)
+option(USE_OPENCL "Build with OpenCL support" OFF)
+
+# USE_CUDA, USE_HIP, and USE_OPENCL shall be mutually exclusive options
+if ((USE_CUDA AND USE_HIP)
+    OR (USE_CUDA AND USE_OPENCL)
+    OR (USE_OPENCL AND USE_HIP))
```

Comment:
should we maybe change to a `USE_ACCEL=(CUDA|HIP|OPENCL)` or can we anticipate that we are capable of including support for different accelerators in the same binary

Reply:
Good suggestion! I will try getting this into this PR or log an issue if not.

Reply:
I have implemented `USE_ACCEL` as part of this PR.